skbase/validate/_named_objects.py

```diff
@@ -157,6 +174,7 @@ def check_sequence_named_objects(
     seq_to_check: Union[Sequence[Tuple[str, BaseObject]], Dict[str, BaseObject]],
```

Comment:
I really don't like this additional amount of non-functional code generated from `overload`, but it's a matter of taste and not a blocker for me...